module/multimedia/src/Infrastructure/Service/DuplicateFilenameSuffixGeneratingService.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/**
+ * Copyright © Ergonode Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\Multimedia\Infrastructure\Service;
+
+class DuplicateFilenameSuffixGeneratingService implements SuffixGeneratingServiceInterface
+{
+    private const MAX_LENGTH = 128;
+
+    public function generateSuffix(string $name, int $iterationIndex): string
+    {
+        $suffix = '('.$iterationIndex.')';
+        if (strlen($name) > (self::MAX_LENGTH - strlen($suffix))) {
```

Comment:
Fairly sure `mb_strlen` should be used. Otherwise, we'd be truncating more than intended in multi-byte characters which are still treated as a single character by Postgre